smart_campus/app/views.py

```diff
@@ -1036,3 +1067,153 @@ def travelplan_delete_page(request, pk):
     travelplan.delete()
 
     return HttpResponseRedirect('/travelplans/')
+
+
+@login_required
+def question_list_page(request):
+    if request.user.is_administrator():
+        stations = Station.objects.all()
+        question_list = Question.objects.all().order_by('id')
+    else:
+        stations = Satation.objects.filter(owner_group=request.user.group)
+        question_list = Question.objects.filter(
+            linked_station__in=stations
+        ).order_by('id')
+
+    paginator = Paginator(question_list, 10)
+
+    page = request.GET.get('page', 1)
+    try:
+        questions = paginator.page(page)
+    except PageNotAnInteger:
+        questions = paginator.page(1)
+    except EmptyPage:
+        questions = paginator.page(paginator.num_pages)
+
+    context = {
+        'questions': questions,
+        'email': request.user.email,
+        'categories': StationCategory.objects.all().order_by('id')
+    }
+
+    return render(request, 'app/question_list_page.html', context)
+
+
+@login_required
+def question_add_page(request):
+
+    if request.method == 'POST':
+        form = QuestionForm(request.POST)
+
+        if form.is_valid():
+            question = form.save()
+
+            choice_contents = request.POST.getlist('choice_contents')
+            answer_order = int(request.POST.get('answer', 1))
```

Comment:
What is this `answer_order`?

Reply:
because the input form can just send the ordered list not a object format like json.
I use the answer_order to indicate which one choice in the sent list is the answer of this question.

I admitted my way is stupid, if you have a much better way, please tell me and I will fix it in no time :)